lib/AST/ASTPrinter.cpp

```diff
@@ -2742,7 +2742,7 @@ void PrintAST::printFunctionParameters(AbstractFunctionDecl *AFD) {
   for (unsigned CurrPattern = 0, NumPatterns = BodyParams.size();
        CurrPattern != NumPatterns; ++CurrPattern) {
     // Be extra careful in the event of printing mal-formed ASTs
-    auto paramListType = parameterListTypes.size() > CurrPattern
+    auto paramListType = CurrPattern < parameterListTypes.size()
```

Comment:
I can't off-by-one. Thanks for putting this in the conventional order so that I don't get confused.